### PR TITLE
created Drivetrain

### PR DESCRIPTION
**Pull request by AmitBrunicki**

### CODE REVIEW

File: src/main/java/frc/robot/RobotMap.java [hunk 1]
```diff
@@ -3,7 +3,10 @@
 public class RobotMap {
 
     public interface CAN {
-
+        int DRIVETRAIN_LEFT_SPARKMAX_1 = -1;
```
Comment (Spikes-2212) [inline src/main/java/frc/robot/RobotMap.java:6]: add a blank line

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    public PIDSettings getDrivePIDSettings() {
+        return drivePIDSettings;
+    }
+
+    public PIDSettings getAnglePIDSettings() {
+        return anglePIDSettings;
+    }
+
+    @Override
+    public void configureDashboard() {
```
Comment (Spikes-2212) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:65]: add resetEncoders method and add it to configureDashboard (also override it to runWhileDisabled)

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import com.spikes2212.command.drivetrains.TankDrivetrain;
+import com.spikes2212.command.drivetrains.smartmotorcontrollerdrivetrain.SparkMaxTankDrivetrain;
+import com.spikes2212.control.PIDSettings;
+import com.spikes2212.dashboard.ChildNamespace;
+import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
+import frc.robot.RobotMap;
+
+import java.util.function.Supplier;
+
+public class Drivetrain extends SparkMaxTankDrivetrain {
```
Comment (Spikes-2212) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:15]: add getters for encoders positions

Comment (Spikes-2212) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:15]: add distance per pulse constant
you need to add constants for gear ratio, wheel diameter (in inches), and inches to centimeters 

after this use `encoder.setConversionFactor` in constructor

Comment (Spikes-2212) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:15]: write a comment to add navx `//@todo add navx`

**Review by Spikes-2212 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 3 above)
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:15]: Add fields with getters that will be used for path following:
`DifferentialDriveOdometry odometry` (update in periodic according to documentation)
`DifferentialDriveKinematics kinematics`
`RamseteController ramseteController`
`Field2d field2d` (no getter, update in periodic according to documentation)

and methods:
`double getLeftSpeed(),
double getRightSpeed(),
Pose2d getPose2d()`
you need to override `configureLoop()`

Also add `FeedForwardSettings` and `TrapezoidProfileSettings`

Add navX json file and add the AHRS, use `Rotation2d` from the AHRS (`gyro.getRotation2d()`)

**Review by OfriRosenbaum — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/RobotMap.java (hunk 1 above)
Comment (AmitBrunicki) [reply src/main/java/frc/robot/RobotMap.java:6]: Why? how do you know? how do you know? why it is important? tell me.

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 2 above)
Comment (AmitBrunicki) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:65]: OK

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 3 above)
Comment (AmitBrunicki) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:15]: easy 🥇

**Pull request closed without merging**